drafts/inlinemacros.md

```diff
@@ -35,8 +34,12 @@ members. Instead, every inline member becomes an overloaded alternative
 of all other members with the same name. Normal overloading resolution
 is applied to pick an inline member over some other member.
 
-Inline definitions only exist at compile-time; no storage is allocated for them and
-no code is generated for them. That means that it is OK to have an inline member that
+If an inline definition doesn't explicitly specify its result type,
+the result type gets inferred according to the usual rules.
```

Comment:
I think that allowing type inference here makes sense, especially given that I've removed the possibility of whitebox expansion at the bottom of the document.